src/iface/ethernet.rs

```diff
@@ -679,6 +703,36 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         }
     }
 
+    #[cfg(feature = "proto-ipv6")]
+    fn process_icmpv6<'frame>(&self, _sockets: &mut SocketSet, ip_repr: IpRepr,
+                              ip_payload: &'frame [u8]) -> Result<Packet<'frame>>
+    {
+        let icmp_packet = Icmpv6Packet::new_checked(ip_payload)?;
+        let checksum_caps = self.device_capabilities.checksum.clone();
```

Comment:
Unrelated to the PR, but do we really need to clone `self.device_capabilities.checksum` everywhere?

Reply:
Borrow checker and closures.